src/views/mfa-verify/PassCodeForm.js

```diff
@@ -87,6 +102,13 @@ define(['okta', 'q', 'views/shared/TextBox'], function (Okta, Q, TextBox) {
         click: function () {
           form.clearErrors();
           this.disable();
+          if (factorType =='call' || factorType == 'sms') {
+            var warning = Okta.loc(factorType == 'call' ? 'factor.call.time.warning' : 'factor.sms.time.warning', 'login');
+            form.clearWarnings();
+            warningTimeout = Util.callAfterTimeout(_.bind(function () {
```

Comment:
we can probably do this at the same time as we change the button text and re-enable it right? In line 124, after saving. Otherwise it could happen that we will see this callout saying `click resend` but the button won't be enabled yet

Reply:
Actually, the time out counter starts at the time the button is clicked and appears as it greys out. If we put this at line 124, that will wait for `Q.Delay` and do another 30s wait. Since they share the same timing variable `API_RATE_LIMIT`, it would appear at the same time. Click Resend and Warning  will appear at the same time.